app/code/Magento/Quote/Api/CartItemRepositoryInterface.php

```diff
@@ -22,15 +22,26 @@ interface CartItemRepositoryInterface
     public function getList($cartId);
 
     /**
-     * Add/update the specified cart item.
+     * Update the specified cart item.
      *
      * @param \Magento\Quote\Api\Data\CartItemInterface $cartItem The item.
      * @return \Magento\Quote\Api\Data\CartItemInterface Item.
      * @throws \Magento\Framework\Exception\NoSuchEntityException The specified cart does not exist.
      * @throws \Magento\Framework\Exception\CouldNotSaveException The specified item could not be saved to the cart.
      * @throws \Magento\Framework\Exception\InputException The specified item or cart is not valid.
      */
-    public function save(\Magento\Quote\Api\Data\CartItemInterface $cartItem);
+    public function update(\Magento\Quote\Api\Data\CartItemInterface $cartItem);
+
+    /**
+     * Add the specified cart item.
+     *
+     * @param \Magento\Quote\Api\Data\CartItemInterface $cartItem The item.
+     * @return \Magento\Quote\Api\Data\CartItemInterface Item.
+     * @throws \Magento\Framework\Exception\NoSuchEntityException The specified cart does not exist.
+     * @throws \Magento\Framework\Exception\CouldNotSaveException The specified item could not be saved to the cart.
+     * @throws \Magento\Framework\Exception\InputException The specified item or cart is not valid.
+     */
+    public function add(\Magento\Quote\Api\Data\CartItemInterface $cartItem);
```

Comment:
Hm... Unfortunately, making such a change will significantly break backward compatibility from the custom extensions perspective, while from the web API perspective it looks ok.
It means that everyone in their custom extensions have to replace `$repository->save($cartItem)` to the `$repository->update($cartItem)` or `$repository->add($cartItem)` depending on situation.

I don't really think this is a good option for fixing this issue.

Reply:
so here we can leave the name of the method.